src/Tools/Source/RunTests/Cache/CachingTestExecutor.cs

```diff
@@ -55,15 +55,19 @@ internal CachingTestExecutor(Options options, ITestExecutor testExecutor, IDataS
         /// <returns></returns>
         private static TestResult Migrate(TestResult testResult)
         {
-            if (string.IsNullOrEmpty(testResult.ResultsFilePath))
```

Comment:
I think your description says that this will no longer be either null or empty.  Maybe assert that?